src/Notify/DependencyInjection/NotifyExtension.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Notify\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\HttpKernel\DependencyInjection\ConfigurableExtension;
+use Symfony\UX\Notify\Twig\NotifyExtension as TwigNotifyExtension;
+use Symfony\WebpackEncoreBundle\Twig\StimulusTwigExtension;
+use Twig\Environment;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @internal
+ */
+final class NotifyExtension extends ConfigurableExtension
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function loadInternal(array $config, ContainerBuilder $container)
+    {
+        if (class_exists(Environment::class) && class_exists(StimulusTwigExtension::class)) {
```

Comment:
Why not make these "hard" dependencies in `composer.json`? My thinking: the main/only thing this bundle provides is the `stream_notifications()` Twig function. If you install this package and are missing either or these two dependencies (especially `StimulusTwigExtension`), it could confuse people to silently not register the new function. So, I think we make these both real dependencies.

(side note, you could argue that Encore shouldn't be needed... just to get the `StimulusTwigExtension` and that's valid. But that's a separate topic).

Reply:
100% agreed!